### PR TITLE
Custom ParaView representation for axis-aligned vtkStructuredGrids

### PR DESCRIPTION
**Pull request by quantumsteve**
Description of work.

This adds a [custom representation](https://www.paraview.org/Wiki/ParaView/Plugin_HowTo#Adding_new_Representations_for_3D_View_using_Plugins_.2A_new_in_version_3.7) based on the `vtkGeometrySliceRepresentation`.  The main change is an `AlignedCutter` class which that speeds up slicing by taking advantage of the consistent bin sizes in a `MDHistoWorkspace`. Other data types like `vtkUnstructuredGrid` fall back to the implementation in `vtkCutter`.

Combined with the patch from [this MR](https://gitlab.kitware.com/paraview/paraview/merge_requests/1850), one should be close to fluid movement with smaller (200x200x200) datasets.
 
**To test:**

* Build the `master` branch with `MAKE_VATES=ON`, load a `MDHistoWorkspace` in the Multislice view. Drag the slices to different positions and observe how long it takes to draw each slice.
* Build this branch and draw load the same `MDHistoWorkspace`. Verify that the view is significantly more responsive.
* Load a `MDEventWorkspace` in the Multislice view and verify there are no issues with the data type.

There is no GitHub issue associated with this pull request.

**Release Notes** 
[here](https://github.com/mantidproject/mantid/pull/20303/commits/d678835b44c80afe87db94a5b297816f73211f1e)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v3.11.0/ui.rst [hunk 1]
```diff
@@ -58,6 +58,7 @@ SliceViewer Improvements
 VSI Improvments
 ---------------
 - ParaView has been updated to to `v5.4.0 <https://blog.kitware.com/paraview-5-4-0-release-notes/>`_.
+- Multislice view uses a custom representation to speed up slicing by taking advantage of the consistent bin sizes in a MDHistoWorkspace.
```
Comment (OwenArnold) [inline docs/source/release/v3.11.0/ui.rst]: Would be nice to have an image of the new representation here.

Comment (peterfpeterson) [inline docs/source/release/v3.11.0/ui.rst]: A link to what a representation is and a second sentence that explains what this actually means to the user would help.

Comment (quantumsteve) [inline docs/source/release/v3.11.0/ui.rst]: done.

**Comment by OwenArnold** [comment]
Looks really good. That image should be used for the release notes in my opinion.

**Review by AndreiSavici — APPROVED** [review]
Seems to be working fast enough :grinning:

**Pull request merged**